[BUG] DWIN ProUI does not compile with latest bugfix-2.0.x

### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

Using the stock configurations from:

Building *with* a BLTouch:
```
Marlin/src/lcd/e3v2/proui/dwin.cpp: In function 'float Tram(uint8_t)':
Marlin/src/lcd/e3v2/proui/dwin.cpp:2344:19: error: 'MESH_MIN_X' was not declared in this scope
 2344 |       LIMIT(xpos, MESH_MIN_X, MESH_MAX_X);
      |                   ^~~~~~~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:2344:31: error: 'MESH_MAX_X' was not declared in this scope
 2344 |       LIMIT(xpos, MESH_MIN_X, MESH_MAX_X);
      |                               ^~~~~~~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:2345:19: error: 'MESH_MIN_Y' was not declared in this scope
 2345 |       LIMIT(ypos, MESH_MIN_Y, MESH_MAX_Y);
      |                   ^~~~~~~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:2345:31: error: 'MESH_MAX_Y' was not declared in this scope
 2345 |       LIMIT(ypos, MESH_MIN_Y, MESH_MAX_Y);
      |                               ^~~~~~~~~~
*** [.pio/build/STM32F103RE_creality/src/src/lcd/e3v2/proui/dwin.cpp.o] Error 1
```

Building *without* a BLTouch:
```
In file included from Marlin/src/lcd/e3v2/proui/../../../inc/MarlinConfigPre.h:37,
                 from Marlin/src/lcd/e3v2/proui/../../../inc/MarlinConfig.h:28,
                 from Marlin/src/lcd/e3v2/proui/dwin.cpp:30:
Marlin/src/lcd/e3v2/proui/dwin.cpp: In function 'void update_variable()':
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:65: error: 'struct HMI_data_t' has no member named 'ManualZOffset'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                                                                 ^~~~~~~~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:621:26: note: in definition of macro 'THIRD'
  621 | #define THIRD(a,b,c,...) c
      |                          ^
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:247:29: note: in expansion of macro '___TERN'
  247 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:246:29: note: in expansion of macro '__TERN'
  246 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:242:29: note: in expansion of macro '_TERN'
  242 | #define TERN(O,A,B)         _TERN(_ENA_1(O),B,A)    // OPTION ? 'A' : 'B'
      |                             ^~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:20: note: in expansion of macro 'TERN'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                    ^~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:803:7: note: in expansion of macro 'BABY_Z_VAR'
  803 |   if (BABY_Z_VAR != _offset) {
      |       ^~~~~~~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:65: error: 'struct HMI_data_t' has no member named 'ManualZOffset'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                                                                 ^~~~~~~~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:621:26: note: in definition of macro 'THIRD'
  621 | #define THIRD(a,b,c,...) c
      |                          ^
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:247:29: note: in expansion of macro '___TERN'
  247 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:246:29: note: in expansion of macro '__TERN'
  246 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:242:29: note: in expansion of macro '_TERN'
  242 | #define TERN(O,A,B)         _TERN(_ENA_1(O),B,A)    // OPTION ? 'A' : 'B'
      |                             ^~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:20: note: in expansion of macro 'TERN'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                    ^~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:804:15: note: in expansion of macro 'BABY_Z_VAR'
  804 |     _offset = BABY_Z_VAR;
      |               ^~~~~~~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp: In function 'void DWIN_Draw_Dashboard()':
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:65: error: 'struct HMI_data_t' has no member named 'ManualZOffset'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                                                                 ^~~~~~~~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:621:26: note: in definition of macro 'THIRD'
  621 | #define THIRD(a,b,c,...) c
      |                          ^
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:247:29: note: in expansion of macro '___TERN'
  247 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:246:29: note: in expansion of macro '__TERN'
  246 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:242:29: note: in expansion of macro '_TERN'
  242 | #define TERN(O,A,B)         _TERN(_ENA_1(O),B,A)    // OPTION ? 'A' : 'B'
      |                             ^~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:20: note: in expansion of macro 'TERN'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                    ^~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:1037:115: note: in expansion of macro 'BABY_Z_VAR'
 1037 |   DWINUI::Draw_Signed_Float(DWIN_FONT_STAT, HMI_data.Indicator_Color,  HMI_data.Background_Color, 2, 2, 204, 417, BABY_Z_VAR);
      |                                                                                                                   ^~~~~~~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp: In function 'void SetZOffset()':
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:65: error: 'struct HMI_data_t' has no member named 'ManualZOffset'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                                                                 ^~~~~~~~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:621:26: note: in definition of macro 'THIRD'
  621 | #define THIRD(a,b,c,...) c
      |                          ^
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:247:29: note: in expansion of macro '___TERN'
  247 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:246:29: note: in expansion of macro '__TERN'
  246 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:242:29: note: in expansion of macro '_TERN'
  242 | #define TERN(O,A,B)         _TERN(_ENA_1(O),B,A)    // OPTION ? 'A' : 'B'
      |                             ^~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:20: note: in expansion of macro 'TERN'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                    ^~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:2001:75: note: in expansion of macro 'BABY_Z_VAR'
 2001 |       babystep.accum = round(planner.settings.axis_steps_per_mm[Z_AXIS] * BABY_Z_VAR);
      |                                                                           ^~~~~~~~~~
In file included from Marlin/src/lcd/e3v2/proui/dwin.cpp:35:
Marlin/src/lcd/e3v2/proui/dwin.cpp: In function 'void Draw_Prepare_Menu()':
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:65: error: 'struct HMI_data_t' has no member named 'ManualZOffset'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                                                                 ^~~~~~~~~~~~~
Marlin/src/lcd/e3v2/proui/menus.h:69:62: note: in definition of macro 'EDIT_ITEM_F'
   69 | #define EDIT_ITEM_F(I,L,V...) MenuItemsAdd(I, GET_TEXT_F(L), V)
      |                                                              ^
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:248:29: note: in expansion of macro 'THIRD'
  248 | #define ___TERN(P,V...)     THIRD(P,V)              // If first argument has a comma, A. Else B.
      |                             ^~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:247:29: note: in expansion of macro '___TERN'
  247 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:246:29: note: in expansion of macro '__TERN'
  246 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin/src/lcd/e3v2/proui/../../../inc/../core/macros.h:242:29: note: in expansion of macro '_TERN'
  242 | #define TERN(O,A,B)         _TERN(_ENA_1(O),B,A)    // OPTION ? 'A' : 'B'
      |                             ^~~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:165:20: note: in expansion of macro 'TERN'
  165 | #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
      |                    ^~~~
Marlin/src/lcd/e3v2/proui/dwin.cpp:2993:87: note: in expansion of macro 'BABY_Z_VAR'
 2993 |         EDIT_ITEM_F(ICON_Zoffset, MSG_ZPROBE_ZOFFSET, onDrawPFloat2Menu, SetZOffset, &BABY_Z_VAR);
      |                                                                                       ^~~~~~~~~~
compilation terminated due to -fmax-errors=5.
*** [.pio/build/STM32F103RE_creality/src/src/lcd/e3v2/proui/dwin.cpp.o] Error 1
Marlin/src/lcd/e3v2/proui/ubl_tools.cpp: In static member function 'static void UBLMeshToolsClass::manual_move(uint8_t, uint8_t, bool)':
Marlin/src/lcd/e3v2/proui/ubl_tools.cpp:125:44: error: 'bedlevel' was not declared in this scope
  125 |     current_position.z = goto_mesh_value ? bedlevel.z_values[mesh_x][mesh_y] : Z_CLEARANCE_BETWEEN_PROBES;
      |                                            ^~~~~~~~
Marlin/src/lcd/e3v2/proui/ubl_tools.cpp:137:44: error: 'bedlevel' was not declared in this scope
  137 |     current_position.z = goto_mesh_value ? bedlevel.z_values[mesh_x][mesh_y] : Z_CLEARANCE_BETWEEN_PROBES;
      |                                            ^~~~~~~~
Marlin/src/lcd/e3v2/proui/ubl_tools.cpp: In static member function 'static float UBLMeshToolsClass::get_max_value()':
Marlin/src/lcd/e3v2/proui/ubl_tools.cpp:146:13: error: 'x' was not declared in this scope
  146 |   GRID_LOOP(x, y) {
      |             ^
Marlin/src/lcd/e3v2/proui/ubl_tools.cpp:146:16: error: 'y' was not declared in this scope; did you mean 'yn'?
  146 |   GRID_LOOP(x, y) {
      |                ^
      |                yn
Marlin/src/lcd/e3v2/proui/ubl_tools.cpp:146:3: error: 'GRID_LOOP' was not declared in this scope
  146 |   GRID_LOOP(x, y) {
      |   ^~~~~~~~~
compilation terminated due to -fmax-errors=5.
*** [.pio/build/STM32F103RE_creality/src/src/lcd/e3v2/proui/ubl_tools.cpp.o] Error 1
```

### Bug Timeline

18/05/2022

### Expected behavior

.

### Actual behavior

.

### Steps to Reproduce

.

### Version of Marlin Firmware

bugfix-2.0.x

### Printer model

Ender 3 V2 + DWIN

### Electronics

_No response_

### Add-ons

_No response_

### Bed Leveling

_No response_

### Your Slicer

_No response_

### Host Software

_No response_

### Additional information & file uploads

_No response_

--- a/Marlin/src/lcd/e3v2/proui/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/proui/dwin.cpp
@@ -162,7 +162,27 @@
 #define DWIN_SCROLL_UPDATE_INTERVAL      SEC_TO_MS(2)
 #define DWIN_REMAIN_TIME_UPDATE_INTERVAL SEC_TO_MS(20)
 
-#define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
+#if HAS_MESH
+  #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
+
+  // These may be overridden in Configuration.h if a smaller area is desired
+  // AUTO_BED_LEVELING_BILINEAR does not define MESH_INSET
+  #ifndef MESH_MIN_X
+    #define MESH_MIN_X (_MAX(X_MIN_BED + PROBING_MARGIN, X_MIN_POS))
+  #endif
+  #ifndef MESH_MIN_Y
+    #define MESH_MIN_Y (_MAX(Y_MIN_BED + PROBING_MARGIN, Y_MIN_POS))
+  #endif
+  #ifndef MESH_MAX_X
+    #define MESH_MAX_X (_MIN(X_MAX_BED - (PROBING_MARGIN), X_MAX_POS))
+  #endif
+  #ifndef MESH_MAX_Y
+    #define MESH_MAX_Y (_MIN(Y_MAX_BED - (PROBING_MARGIN), Y_MAX_POS))
+  #endif
+#else
+  float z_offset = 0;
+  #define BABY_Z_VAR z_offset
+#endif
 
 // Structs
 HMI_value_t HMI_value;
@@ -200,11 +220,23 @@ uint8_t index_file     = MROWS;
 
 bool hash_changed = true; // Flag to know if message status was changed
 
-constexpr float max_feedrate_edit_values[] = MAX_FEEDRATE_EDIT_VALUES;
-constexpr float max_acceleration_edit_values[] = MAX_ACCEL_EDIT_VALUES;
+#if ENABLED(LIMITED_MAX_FR_EDITING)
+  constexpr float max_feedrate_edit_values[] = MAX_FEEDRATE_EDIT_VALUES;
+#else
+  constexpr float max_feedrate_edit_values[] = { 1000, 1000, 10, 50 };
+#endif
+#if ENABLED(LIMITED_MAX_ACCEL_EDITING)
+  constexpr float max_acceleration_edit_values[] = MAX_ACCEL_EDIT_VALUES;
+#else
+  constexpr float max_acceleration_edit_values[] = { 1000, 1000, 200, 2000 };
+#endif
 
 #if HAS_CLASSIC_JERK
-  constexpr float max_jerk_edit_values[] = MAX_JERK_EDIT_VALUES;
+  #if ENABLED(LIMITED_JERK_EDITING)
+    constexpr float max_jerk_edit_values[] = MAX_JERK_EDIT_VALUES;
+  #else
+    constexpr float max_jerk_edit_values[] = { DEFAULT_XJERK * 2, DEFAULT_YJERK * 2, DEFAULT_ZJERK * 2, DEFAULT_EJERK * 2 };
+  #endif
 #endif
 
 static uint8_t _percent_done = 0;
@@ -815,6 +847,8 @@ void update_variable() {
       else
         DWINUI::Draw_Icon(ICON_Zoffset, 187, 416);
     }
+  #else
+    DWINUI::Draw_Icon(ICON_Zoffset, 187, 416);
   #endif
 
   _draw_xyz_position(false);
@@ -3606,18 +3640,19 @@ void Draw_Steps_Menu() {
     void SetBedLevT() { SetPIntOnClick(MIN_BEDTEMP, MAX_BEDTEMP); }
   #endif
 
-  uint8_t mesh_x = 0;
-  uint8_t mesh_y = 0;
-  #define Z_OFFSET_MIN -3
-  #define Z_OFFSET_MAX  3
-
-  void LiveEditMesh() { ((MenuItemPtrClass*)EditZValueItem)->value = &bedlevel.z_values[HMI_value.Select ? mesh_x : MenuData.Value][HMI_value.Select ? MenuData.Value : mesh_y]; EditZValueItem->redraw(); }
-  void ApplyEditMeshX() { mesh_x = MenuData.Value; }
-  void SetEditMeshX() { HMI_value.Select = 0; SetIntOnClick(0, GRID_MAX_POINTS_X - 1, mesh_x, ApplyEditMeshX, LiveEditMesh); }
-  void ApplyEditMeshY() { mesh_y = MenuData.Value; }
-  void SetEditMeshY() { HMI_value.Select = 1; SetIntOnClick(0, GRID_MAX_POINTS_Y - 1, mesh_y, ApplyEditMeshY, LiveEditMesh); }
-  void SetEditZValue() { SetPFloatOnClick(Z_OFFSET_MIN, Z_OFFSET_MAX, 3); }
-
+  #if ENABLED(MESH_EDIT_MENU)
+    uint8_t mesh_x = 0;
+    uint8_t mesh_y = 0;
+    #define Z_OFFSET_MIN -3
+    #define Z_OFFSET_MAX  3
+  
+    void LiveEditMesh() { ((MenuItemPtrClass*)EditZValueItem)->value = &bedlevel.z_values[HMI_value.Select ? mesh_x : MenuData.Value][HMI_value.Select ? MenuData.Value : mesh_y]; EditZValueItem->redraw(); }
+    void ApplyEditMeshX() { mesh_x = MenuData.Value; }
+    void SetEditMeshX() { HMI_value.Select = 0; SetIntOnClick(0, GRID_MAX_POINTS_X - 1, mesh_x, ApplyEditMeshX, LiveEditMesh); }
+    void ApplyEditMeshY() { mesh_y = MenuData.Value; }
+    void SetEditMeshY() { HMI_value.Select = 1; SetIntOnClick(0, GRID_MAX_POINTS_Y - 1, mesh_y, ApplyEditMeshY, LiveEditMesh); }
+    void SetEditZValue() { SetPFloatOnClick(Z_OFFSET_MIN, Z_OFFSET_MAX, 3); }
+  #endif
 #endif
 
 #if ENABLED(AUTO_BED_LEVELING_UBL)

--- a/Marlin/src/lcd/e3v2/proui/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/proui/dwin.cpp
@@ -195,21 +195,6 @@
 
 #if HAS_MESH
   #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
-
-  // These may be overridden in Configuration.h if a smaller area is desired
-  // AUTO_BED_LEVELING_BILINEAR does not define MESH_INSET
-  #ifndef MESH_MIN_X
-    #define MESH_MIN_X (_MAX(X_MIN_BED + PROBING_MARGIN, X_MIN_POS))
-  #endif
-  #ifndef MESH_MIN_Y
-    #define MESH_MIN_Y (_MAX(Y_MIN_BED + PROBING_MARGIN, Y_MIN_POS))
-  #endif
-  #ifndef MESH_MAX_X
-    #define MESH_MAX_X (_MIN(X_MAX_BED - (PROBING_MARGIN), X_MAX_POS))
-  #endif
-  #ifndef MESH_MAX_Y
-    #define MESH_MAX_Y (_MIN(Y_MAX_BED - (PROBING_MARGIN), Y_MAX_POS))
-  #endif
 #else
   float z_offset = 0;
   #define BABY_Z_VAR z_offset
@@ -251,23 +236,28 @@ uint8_t index_file     = MROWS;
 
 bool hash_changed = true; // Flag to know if message status was changed
 
-#if ENABLED(LIMITED_MAX_FR_EDITING)
-  constexpr float max_feedrate_edit_values[] = MAX_FEEDRATE_EDIT_VALUES;
-#else
-  constexpr float max_feedrate_edit_values[] = { 1000, 1000, 10, 50 };
-#endif
-#if ENABLED(LIMITED_MAX_ACCEL_EDITING)
-  constexpr float max_acceleration_edit_values[] = MAX_ACCEL_EDIT_VALUES;
-#else
-  constexpr float max_acceleration_edit_values[] = { 1000, 1000, 200, 2000 };
-#endif
-
-#if HAS_CLASSIC_JERK
-  #if ENABLED(LIMITED_JERK_EDITING)
-    constexpr float max_jerk_edit_values[] = MAX_JERK_EDIT_VALUES;
+constexpr float max_feedrate_edit_values[] =
+  #ifdef MAX_FEEDRATE_EDIT_VALUES
+    MAX_FEEDRATE_EDIT_VALUES
   #else
-    constexpr float max_jerk_edit_values[] = { DEFAULT_XJERK * 2, DEFAULT_YJERK * 2, DEFAULT_ZJERK * 2, DEFAULT_EJERK * 2 };
+    { 1000, 1000, 10, 50 }
   #endif
+;
+constexpr float max_acceleration_edit_values[] =
+  #ifdef MAX_ACCEL_EDIT_VALUES
+    MAX_ACCEL_EDIT_VALUES
+  #else
+    { 1000, 1000, 200, 2000 }
+  #endif
+;
+#if HAS_CLASSIC_JERK
+  constexpr float max_jerk_edit_values[] =
+    #ifdef MAX_JERK_EDIT_VALUES
+      MAX_JERK_EDIT_VALUES
+    #else
+      { DEFAULT_XJERK * 2, DEFAULT_YJERK * 2, DEFAULT_ZJERK * 2, DEFAULT_EJERK * 2 }
+    #endif
+  ;
 #endif
 
 static uint8_t _percent_done = 0;
@@ -2357,48 +2347,45 @@ void ApplyFlow() { planner.refresh_e_factor(0); }
 void SetFlow() { SetPIntOnClick(MIN_PRINT_FLOW, MAX_PRINT_FLOW, ApplyFlow); }
 
 // Bed Tramming
-TERN(HAS_BED_PROBE, float, void) Tram(uint8_t point) {
-  char cmd[100] = "";
-  #if HAS_BED_PROBE
+
+void TramXY(const uint8_t point, const float &margin, float &x, float &y) {
+  switch (point) {
+    case 0:
+      LCD_MESSAGE(MSG_LEVBED_FL);
+      x = y = margin;
+      break;
+    case 1:
+      LCD_MESSAGE(MSG_LEVBED_FR);
+      x = X_BED_SIZE - margin; y = margin;
+      break;
+    case 2:
+      LCD_MESSAGE(MSG_LEVBED_BR);
+      x = X_BED_SIZE - margin; y = Y_BED_SIZE - margin;
+      break;
+    case 3:
+      LCD_MESSAGE(MSG_LEVBED_BL);
+      x = margin; y = Y_BED_SIZE - margin;
+      break;
+    case 4:
+      LCD_MESSAGE(MSG_LEVBED_C);
+      x = X_CENTER; y = Y_CENTER;
+      break;
+  }
+}
+
+#if HAS_BED_PROBE
+
+  float Tram(const uint8_t point) {
+    char cmd[100] = "";
     static bool inLev = false;
     float xpos = 0, ypos = 0, zval = 0, margin = 0;
     char str_1[6] = "", str_2[6] = "", str_3[6] = "";
     if (inLev) return NAN;
     margin = HMI_data.FullManualTramming ? 30 : PROBING_MARGIN;
-  #else
-    int16_t xpos = 0, ypos = 0;
-    int16_t margin = 30;
-  #endif
 
-  switch (point) {
-    case 0:
-      LCD_MESSAGE(MSG_LEVBED_FL);
-      xpos = ypos = margin;
-      break;
-    case 1:
-      LCD_MESSAGE(MSG_LEVBED_FR);
-      xpos = X_BED_SIZE - margin; ypos = margin;
-      break;
-    case 2:
-      LCD_MESSAGE(MSG_LEVBED_BR);
-      xpos = X_BED_SIZE - margin; ypos = Y_BED_SIZE - margin;
-      break;
-    case 3:
-      LCD_MESSAGE(MSG_LEVBED_BL);
-      xpos = margin; ypos = Y_BED_SIZE - margin;
-      break;
-    case 4:
-      LCD_MESSAGE(MSG_LEVBED_C);
-      xpos = X_BED_SIZE / 2; ypos = Y_BED_SIZE / 2;
-      break;
-  }
-
-  planner.synchronize();
-
-  #if HAS_BED_PROBE
+    TramXY(point, margin, xpos, ypos);
 
     if (HMI_data.FullManualTramming) {
-      planner.synchronize();
       sprintf_P(cmd, PSTR("M420S0\nG28O\nG90\nG0Z5F300\nG0X%sY%sF5000\nG0Z0F300"),
         dtostrf(xpos, 1, 1, str_1),
         dtostrf(ypos, 1, 1, str_2)
@@ -2406,6 +2393,20 @@ TERN(HAS_BED_PROBE, float, void) Tram(uint8_t point) {
       queue.inject(cmd);
     }
     else {
+      // AUTO_BED_LEVELING_BILINEAR does not define MESH_INSET
+      #ifndef MESH_MIN_X
+        #define MESH_MIN_X (_MAX(X_MIN_BED + PROBING_MARGIN, X_MIN_POS))
+      #endif
+      #ifndef MESH_MIN_Y
+        #define MESH_MIN_Y (_MAX(Y_MIN_BED + PROBING_MARGIN, Y_MIN_POS))
+      #endif
+      #ifndef MESH_MAX_X
+        #define MESH_MAX_X (_MIN(X_MAX_BED - (PROBING_MARGIN), X_MAX_POS))
+      #endif
+      #ifndef MESH_MAX_Y
+        #define MESH_MAX_Y (_MIN(Y_MAX_BED - (PROBING_MARGIN), Y_MAX_POS))
+      #endif
+
       LIMIT(xpos, MESH_MIN_X, MESH_MAX_X);
       LIMIT(ypos, MESH_MIN_Y, MESH_MAX_Y);
       probe.stow();
@@ -2425,14 +2426,20 @@ TERN(HAS_BED_PROBE, float, void) Tram(uint8_t point) {
       inLev = false;
     }
     return zval;
+  }
 
-  #else // !HAS_BED_PROBE
+#else
 
-    sprintf_P(cmd, PSTR("M420S0\nG28O\nG90\nG0Z5F300\nG0X%iY%iF5000\nG0Z0F300"), xpos, ypos);
+  void Tram(const uint8_t point) {
+    float xpos = 0, ypos = 0, margin = 30;
+    TramXY(point, margin, xpos, ypos);
+
+    char cmd[100] = "", str_1[6] = "", str_2[6] = "";
+    sprintf_P(cmd, PSTR("M420S0\nG28O\nG90\nG0Z5F300\nG0X%sY%sF5000\nG0Z0F300"), dtostrf(xpos, 1, 1, str_1), dtostrf(ypos, 1, 1, str_2));
     queue.inject(cmd);
+  }
 
-  #endif
-}
+#endif
 
 void TramFL() { Tram(0); }
 void TramFR() { Tram(1); }
@@ -3672,11 +3679,10 @@ void Draw_Steps_Menu() {
   #endif
 
   #if ENABLED(MESH_EDIT_MENU)
-    uint8_t mesh_x = 0;
-    uint8_t mesh_y = 0;
+    uint8_t mesh_x = 0, mesh_y = 0;
     #define Z_OFFSET_MIN -3
     #define Z_OFFSET_MAX  3
-  
+
     void LiveEditMesh() { ((MenuItemPtrClass*)EditZValueItem)->value = &bedlevel.z_values[HMI_value.Select ? mesh_x : MenuData.Value][HMI_value.Select ? MenuData.Value : mesh_y]; EditZValueItem->redraw(); }
     void ApplyEditMeshX() { mesh_x = MenuData.Value; }
     void SetEditMeshX() { HMI_value.Select = 0; SetIntOnClick(0, GRID_MAX_POINTS_X - 1, mesh_x, ApplyEditMeshX, LiveEditMesh); }

--- a/Marlin/src/lcd/e3v2/proui/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/proui/dwin.cpp
@@ -31,6 +31,37 @@
 
 #if ENABLED(DWIN_LCD_PROUI)
 
+#if DISABLED(LIMITED_MAX_FR_EDITING)
+  #warning "LIMITED_MAX_FR_EDITING is recommended with ProUI."
+#endif
+#if DISABLED(LIMITED_MAX_ACCEL_EDITING)
+  #warning "LIMITED_MAX_ACCEL_EDITING is recommended with ProUI."
+#endif
+#if ENABLED(CLASSIC_JERK) && DISABLED(LIMITED_JERK_EDITING)
+  #warning "LIMITED_JERK_EDITING is recommended with ProUI."
+#endif
+#if DISABLED(INDIVIDUAL_AXIS_HOMING_SUBMENU)
+  #warning "INDIVIDUAL_AXIS_HOMING_SUBMENU is recommended with ProUI."
+#endif
+#if DISABLED(LCD_SET_PROGRESS_MANUALLY)
+  #warning "LCD_SET_PROGRESS_MANUALLY is recommended with ProUI."
+#endif
+#if DISABLED(STATUS_MESSAGE_SCROLLING)
+  #warning "STATUS_MESSAGE_SCROLLING is recommended with ProUI."
+#endif
+#if DISABLED(BAUD_RATE_GCODE)
+  #warning "BAUD_RATE_GCODE is recommended with ProUI."
+#endif
+#if DISABLED(SOUND_MENU_ITEM)
+  #warning "SOUND_MENU_ITEM is recommended with ProUI."
+#endif
+#if DISABLED(PRINTCOUNTER)
+  #warning "PRINTCOUNTER is recommended with ProUI."
+#endif
+#if HAS_MESH && DISABLED(MESH_EDIT_MENU)
+  #warning "MESH_EDIT_MENU is recommended with ProUI."
+#endif
+
 #include "dwin.h"
 #include "menus.h"
 #include "dwin_popup.h"

--- a/Marlin/src/lcd/e3v2/proui/dwin.h
+++ b/Marlin/src/lcd/e3v2/proui/dwin.h
@@ -238,9 +238,7 @@ void Draw_Tramming_Menu();
 #if HAS_BED_PROBE
   void Draw_ProbeSet_Menu();
 #endif
-#if HAS_FILAMENT_SENSOR
-  void Draw_FilSet_Menu();
-#endif
+void Draw_FilSet_Menu();
 #if ENABLED(NOZZLE_PARK_FEATURE)
   void Draw_ParkPos_Menu();
 #endif

--- a/Marlin/src/lcd/e3v2/proui/dwin_defines.h
+++ b/Marlin/src/lcd/e3v2/proui/dwin_defines.h
@@ -40,34 +40,37 @@
 #endif
 
 #if DISABLED(LIMITED_MAX_FR_EDITING)
-  #error "LIMITED_MAX_FR_EDITING is required with ProUI."
+  #warning "LIMITED_MAX_FR_EDITING can be enabled with ProUI."
 #endif
 #if DISABLED(LIMITED_MAX_ACCEL_EDITING)
-  #error "LIMITED_MAX_ACCEL_EDITING is required with ProUI."
+  #warning "LIMITED_MAX_ACCEL_EDITING can be enabled with ProUI."
 #endif
 #if ENABLED(CLASSIC_JERK) && DISABLED(LIMITED_JERK_EDITING)
-  #error "LIMITED_JERK_EDITING is required with ProUI."
+  #warning "LIMITED_JERK_EDITING can be enabled with ProUI."
 #endif
 #if DISABLED(FILAMENT_RUNOUT_SENSOR)
-  #error "FILAMENT_RUNOUT_SENSOR is required with ProUI."
+  #warning "FILAMENT_RUNOUT_SENSOR can be enabled with ProUI."
 #endif
 #if DISABLED(INDIVIDUAL_AXIS_HOMING_SUBMENU)
-  #error "INDIVIDUAL_AXIS_HOMING_SUBMENU is required with ProUI."
+  #warning "INDIVIDUAL_AXIS_HOMING_SUBMENU can be enabled with ProUI."
 #endif
 #if DISABLED(LCD_SET_PROGRESS_MANUALLY)
-  #error "LCD_SET_PROGRESS_MANUALLY is required with ProUI."
+  #warning "LCD_SET_PROGRESS_MANUALLY can be enabled with ProUI."
 #endif
 #if DISABLED(STATUS_MESSAGE_SCROLLING)
-  #error "STATUS_MESSAGE_SCROLLING is required with ProUI."
+  #warning "STATUS_MESSAGE_SCROLLING can be enabled with ProUI."
 #endif
 #if DISABLED(BAUD_RATE_GCODE)
-  #error "BAUD_RATE_GCODE is required with ProUI."
+  #warning "BAUD_RATE_GCODE can be enabled with ProUI."
 #endif
 #if DISABLED(SOUND_MENU_ITEM)
-  #error "SOUND_MENU_ITEM is required with ProUI."
+  #warning "SOUND_MENU_ITEM can be enabled with ProUI."
 #endif
 #if DISABLED(PRINTCOUNTER)
-  #error "PRINTCOUNTER is required with ProUI."
+  #warning "PRINTCOUNTER can be enabled with ProUI."
+#endif
+#if DISABLED(MESH_EDIT_MENU)
+  #warning "MESH_EDIT_MENU can be enabled with ProUI."
 #endif
 
 #include "../common/dwin_color.h"

--- a/Marlin/src/lcd/e3v2/proui/dwin_defines.h
+++ b/Marlin/src/lcd/e3v2/proui/dwin_defines.h
@@ -39,40 +39,6 @@
   #define DASH_REDRAW 1
 #endif
 
-#if DISABLED(LIMITED_MAX_FR_EDITING)
-  #warning "LIMITED_MAX_FR_EDITING can be enabled with ProUI."
-#endif
-#if DISABLED(LIMITED_MAX_ACCEL_EDITING)
-  #warning "LIMITED_MAX_ACCEL_EDITING can be enabled with ProUI."
-#endif
-#if ENABLED(CLASSIC_JERK) && DISABLED(LIMITED_JERK_EDITING)
-  #warning "LIMITED_JERK_EDITING can be enabled with ProUI."
-#endif
-#if DISABLED(FILAMENT_RUNOUT_SENSOR)
-  #warning "FILAMENT_RUNOUT_SENSOR can be enabled with ProUI."
-#endif
-#if DISABLED(INDIVIDUAL_AXIS_HOMING_SUBMENU)
-  #warning "INDIVIDUAL_AXIS_HOMING_SUBMENU can be enabled with ProUI."
-#endif
-#if DISABLED(LCD_SET_PROGRESS_MANUALLY)
-  #warning "LCD_SET_PROGRESS_MANUALLY can be enabled with ProUI."
-#endif
-#if DISABLED(STATUS_MESSAGE_SCROLLING)
-  #warning "STATUS_MESSAGE_SCROLLING can be enabled with ProUI."
-#endif
-#if DISABLED(BAUD_RATE_GCODE)
-  #warning "BAUD_RATE_GCODE can be enabled with ProUI."
-#endif
-#if DISABLED(SOUND_MENU_ITEM)
-  #warning "SOUND_MENU_ITEM can be enabled with ProUI."
-#endif
-#if DISABLED(PRINTCOUNTER)
-  #warning "PRINTCOUNTER can be enabled with ProUI."
-#endif
-#if DISABLED(MESH_EDIT_MENU)
-  #warning "MESH_EDIT_MENU can be enabled with ProUI."
-#endif
-
 #include "../common/dwin_color.h"
 #if ENABLED(LED_CONTROL_MENU)
   #include "../../../feature/leds/leds.h"

--- a/Marlin/src/lcd/e3v2/proui/ubl_tools.cpp
+++ b/Marlin/src/lcd/e3v2/proui/ubl_tools.cpp
@@ -23,7 +23,7 @@
 
 #include "../../../inc/MarlinConfigPre.h"
 
-#if BOTH(DWIN_LCD_PROUI, HAS_MESH)
+#if BOTH(DWIN_LCD_PROUI, AUTO_BED_LEVELING_UBL)
 
 #include "ubl_tools.h"
 #include "../../marlinui.h"
@@ -249,4 +249,4 @@ bool UBLMeshToolsClass::validate() {
   }
 #endif
 
-#endif // DWIN_LCD_PROUI
+#endif // DWIN_LCD_PROUI && AUTO_BED_LEVELING_UBL

--- a/Marlin/src/lcd/e3v2/proui/ubl_tools.cpp
+++ b/Marlin/src/lcd/e3v2/proui/ubl_tools.cpp
@@ -22,10 +22,10 @@
  */
 
 #include "../../../inc/MarlinConfigPre.h"
+
+#if BOTH(DWIN_LCD_PROUI, HAS_MESH)
+
 #include "ubl_tools.h"
-
-#if ENABLED(DWIN_LCD_PROUI)
-
 #include "../../marlinui.h"
 #include "../../../core/types.h"
 #include "dwin.h"

